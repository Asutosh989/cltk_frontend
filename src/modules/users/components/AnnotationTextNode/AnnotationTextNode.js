import React from 'react';
import PropTypes from 'prop-types';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import Utils from '../../../../lib/util';

class AnnotationTextNode extends React.Component {
	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	}

	getTextLocation() {
		const text = this.props.text;
		let location = '';
		let textN = '';

		if ('n_1' in text) {
			location += text.n_1;
			textN = text.n_1;
		}
		if ('n_2' in text) {
			location += `.${text.n_2}`;
			textN = text.n_2;
		}
		if ('n_3' in text) {
			location += `.${text.n_3}`;
			textN = text.n_3;
		}
		if ('n_4' in text) {
			location += `.${text.n_4}`;
			textN = text.n_4;
		}
		if ('n_5' in text) {
			location += `.${text.n_5}`;
			textN = text.n_5;
		}

		return {
			location,
			textN,
		};
	}

	handleClick() {

	}


	render() {
		//const text = this.props.text;
		const { annotation } = this.props;
		let textClasses = 'text-node bookmark-text-node annotation-text-node clearfix';
		// const textLocation = this.getTextLocation();
		const textLocation = '';
		let workTitle = '';
		let link = '';

		if (this.props.work) {
			workTitle = this.props.work.english_title;
			link = `/works/${this.props.work._id}/${this.props.work.slug}?location=${textLocation.location}`;
		}

		if (this.props.isOdd) {
			textClasses = `${textClasses} bookmark-text-node--odd`;
		}

		if ((parseInt(textLocation.textN, 10) % 5) === 0) {
			textClasses = `${textClasses} show-number`;
		}

		return (
			<a
				className={textClasses}
				data-id={annotation._id}
				//data-loc={textLocation.location}
				href={link}
			>
				<div className="text-left-header">
					<h2 className="section-numbering">{Utils.trunc(workTitle, 40)} {textLocation.location}</h2>
				</div>

				<p
					className="text-html"
					ref={(ref) => {
						this.anchorEl = ref;
						return ref;
					}}
				>
					{annotation.content && annotation.content.length ?
						<span>{annotation.content}</span>
					:
						<span>[ . . . ]</span>
					}
				</p>

			</a>
		);
	}
};

AnnotationTextNode.childContextTypes = {
	muiTheme: PropTypes.object.isRequired,
};

AnnotationTextNode.propTypes = {
	annotation: PropTypes.object.isRequired,
	isOdd: PropTypes.bool,
};

export default AnnotationTextNode;
