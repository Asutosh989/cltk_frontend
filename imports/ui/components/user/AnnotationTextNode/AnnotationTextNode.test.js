
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import AnnotationTextNode from './AnnotationTextNode';

describe('AnnotationTextNode', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<AnnotationTextNode />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
