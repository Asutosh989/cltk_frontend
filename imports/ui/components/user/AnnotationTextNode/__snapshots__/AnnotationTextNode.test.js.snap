// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotationTextNode renders correctly 1`] = `
<a
  className="text-node bookmark-text-node annotation-text-node clearfix"
  data-id="testId"
  href=""
>
  <div
    className="text-left-header"
  >
    <h2
      className="section-numbering"
    >
      
       
    </h2>
  </div>
  <p
    className="text-html"
  >
    <span>
      [ . . . ]
    </span>
  </p>
</a>
`;
